monkey/monkey_island/cc/services/infection_lifecycle.py

```diff
@@ -16,42 +14,60 @@
 logger = logging.getLogger(__name__)
 
 
-class InfectionLifecycle:
-    @staticmethod
-    def kill_all():
-        mongo.db.monkey.update(
-            {"dead": False},
-            {"$set": {"config.alive": False, "modifytime": datetime.now()}},
-            upsert=False,
-            multi=True,
-        )
-        logger.info("Kill all monkeys was called")
-        return jsonify(status="OK")
-
-    @staticmethod
-    def get_completed_steps():
-        is_any_exists = NodeService.is_any_monkey_exists()
-        infection_done = NodeService.is_monkey_finished_running()
-
-        if infection_done:
-            InfectionLifecycle._on_finished_infection()
-            report_done = ReportService.is_report_generated()
-        else:  # Infection is not done
-            report_done = False
-
-        return dict(
-            run_server=True,
-            run_monkey=is_any_exists,
-            infection_done=infection_done,
-            report_done=report_done,
-        )
-
-    @staticmethod
-    def _on_finished_infection():
-        # Checking is_report_being_generated here, because we don't want to wait to generate a
-        # report; rather,
-        # we want to skip and reply.
-        if not is_report_being_generated() and not ReportService.is_latest_report_exists():
-            safe_generate_reports()
-        if ConfigService.is_test_telem_export_enabled() and not TestTelemStore.TELEMS_EXPORTED:
-            TestTelemStore.export_telems()
+def set_stop_all(time: float):
+    for monkey in Monkey.objects():
+        monkey.config.alive = False
+        monkey.save()
+    agent_controls = AgentControls.objects.first()
+    agent_controls.last_stop_all = time
+    agent_controls.save()
+
+
+def should_agent_die(guid: int) -> bool:
+    monkey = Monkey.objects(guid=guid).first()
+    return _is_monkey_marked_dead(monkey) or _is_monkey_killed_manually()
+
+
+def _is_monkey_marked_dead(monkey: Monkey) -> bool:
+    return monkey.config.alive
+
+
+def _is_monkey_killed_manually(monkey: Monkey) -> bool:
+    if monkey.has_parent():
+        launch_timestamp = monkey.get_parent().launch_time
+    else:
+        launch_timestamp = monkey.launch_time
+    kill_timestamp = AgentControls.objects.first().last_stop_all
+    return int(kill_timestamp) >= int(launch_timestamp)
+
+
+def init_agent_controls():
+    AgentControls().save()
```

Comment:
Maybe this function should live in the database service, instead of in this service.